packages/core/src/config/parse.ts

```diff
@@ -1958,7 +1958,10 @@ export const assertValidContracts = (
         }
       }
 
-      if (!contractConfig.unsafeAllowEmptyPush) {
+      if (
+        !contractConfig.unsafeAllowEmptyPush &&
+        contractConfig.kind !== 'no-proxy'
```

Comment:
What's the purpose of this change?

Reply:
It's not related to the main purpose of this PR. It just skips the `unsafeAllowEmptyPush` check for non-proxied contracts. This is because `push()` can only be dangerous in an upgradeable contract.